Source/carthage/Formatting.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Formatting.swift
+//  Carthage
+//
+//  Created by J.D. Healy on 1/29/15.
+//  Copyright (c) 2015 Carthage. All rights reserved.
+//
+
+import Commandant
+import Foundation
+import LlamaKit
+import PrettyColors
+import ReactiveCocoa
+
+/// Wraps or passes through a string.
+func wrap(colorful: Bool)(wrap: Color.Wrap)(string: String) -> String {
+	return colorful ? wrap.wrap(string) : string
+}
+
+/// Information about the possible parent terminal.
+internal struct Terminal {
+	static var term: String? {
+		return getEnvironmentVariable("TERM").value()
+	}
+	static var isDumb: Bool {
+		return (Terminal.term?.lowercaseString as NSString?)?.isEqualToString("dumb") ?? false
+	}
+	static var isTTY: Bool {
+		return isatty(STDOUT_FILENO) == 1
```

Comment:
It doesn't matter what the source says—in C, 0 is false, and anything non-zero is true. We should honor that convention.